README.md

```diff
@@ -286,6 +286,14 @@ The initial window position and size may be specified:
 scrcpy --window-x 100 --window-y 100 --window-width 800 --window-height 600
 ```
 
+#### Lock orientation
+
+Keep the window in landscape orientation:
+
+```bash
+scrcpy --orientation 3
```

Comment:
Are you using the `dev` branch? This is not merged into `master` yet.

Cheers

Reply:
no, any guide for build exe for win10 64bit?

Reply:
https://github.com/Genymobile/scrcpy/blob/master/BUILD.md

Reply:
debian@debian:~$ sudo apt install mingw-w64 mingw-w64-tools
[sudo] password for debian: 
debian is not in the sudoers file.  This incident will be reported.
debian@debian:~$ su
Password: 
root@debian:/home/debian# sudo apt install mingw-w64 mingw-w64-tools
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  binutils-mingw-w64-i686 binutils-mingw-w64-x86-64 g++-mingw-w64
  g++-mingw-w64-i686 g++-mingw-w64-x86-64 gcc-mingw-w64 gcc-mingw-w64-base
  gcc-mingw-w64-i686 gcc-mingw-w64-x86-64 gfortran-mingw-w64
  gfortran-mingw-w64-i686 gfortran-mingw-w64-x86-64 gnat-mingw-w64
  gnat-mingw-w64-base gnat-mingw-w64-i686 gnat-mingw-w64-x86-64
  mingw-w64-common mingw-w64-i686-dev mingw-w64-x86-64-dev pkg-config
Suggested packages:
  gcc-6-locales wine wine64
The following NEW packages will be installed:
  binutils-mingw-w64-i686 binutils-mingw-w64-x86-64 g++-mingw-w64
  g++-mingw-w64-i686 g++-mingw-w64-x86-64 gcc-mingw-w64 gcc-mingw-w64-base
  gcc-mingw-w64-i686 gcc-mingw-w64-x86-64 gfortran-mingw-w64
  gfortran-mingw-w64-i686 gfortran-mingw-w64-x86-64 gnat-mingw-w64
  gnat-mingw-w64-base gnat-mingw-w64-i686 gnat-mingw-w64-x86-64 mingw-w64
  mingw-w64-common mingw-w64-i686-dev mingw-w64-tools mingw-w64-x86-64-dev
  pkg-config
0 upgraded, 22 newly installed, 0 to remove and 0 not upgraded.
Need to get 209 MB of archives.
After this operation, 1,176 MB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://ftp.us.debian.org/debian stretch/main amd64 pkg-config amd64 0.29-4+b1 [63.3 kB]
Get:2 http://ftp.us.debian.org/debian stretch/main amd64 binutils-mingw-w64-i686 amd64 2.28-5+7.4+b4 [2,242 kB]
Get:3 http://ftp.us.debian.org/debian stretch/main amd64 binutils-mingw-w64-x86-64 amd64 2.28-5+7.4+b4 [2,467 kB]
Get:4 http://ftp.us.debian.org/debian stretch/main amd64 mingw-w64-common all 5.0.1-1 [4,764 kB]
Get:5 http://ftp.us.debian.org/debian stretch/main amd64 mingw-w64-i686-dev all 5.0.1-1 [2,218 kB]
Get:6 http://ftp.us.debian.org/debian stretch/main amd64 gcc-mingw-w64-base amd64 6.3.0-18+19.3+b3 [170 kB]
Get:7 http://ftp.us.debian.org/debian stretch/main amd64 gcc-mingw-w64-i686 amd64 6.3.0-18+19.3+b3 [32.6 MB]
Get:8 http://ftp.us.debian.org/debian stretch/main amd64 g++-mingw-w64-i686 amd64 6.3.0-18+19.3+b3 [21.6 MB]
Get:9 http://ftp.us.debian.org/debian stretch/main amd64 mingw-w64-x86-64-dev all 5.0.1-1 [3,342 kB]
Get:10 http://ftp.us.debian.org/debian stretch/main amd64 gcc-mingw-w64-x86-64 amd64 6.3.0-18+19.3+b3 [32.7 MB]
Get:11 http://ftp.us.debian.org/debian stretch/main amd64 g++-mingw-w64-x86-64 amd64 6.3.0-18+19.3+b3 [22.4 MB]
Get:12 http://ftp.us.debian.org/debian stretch/main amd64 g++-mingw-w64 all 6.3.0-14+19.3 [169 kB]
Get:13 http://ftp.us.debian.org/debian stretch/main amd64 gcc-mingw-w64 all 6.3.0-14+19.3 [169 kB]
Get:14 http://ftp.us.debian.org/debian stretch/main amd64 gfortran-mingw-w64-i686 amd64 6.3.0-18+19.3+b3 [15.7 MB]
Get:15 http://ftp.us.debian.org/debian stretch/main amd64 gfortran-mingw-w64-x86-64 amd64 6.3.0-18+19.3+b3 [16.6 MB]
Get:16 http://ftp.us.debian.org/debian stretch/main amd64 gfortran-mingw-w64 all 6.3.0-14+19.3 [169 kB]
Get:17 http://ftp.us.debian.org/debian stretch/main amd64 gnat-mingw-w64-base amd64 6.3.0-18+19.4+b3 [169 kB]
Get:18 http://ftp.us.debian.org/debian stretch/main amd64 gnat-mingw-w64-i686 amd64 6.3.0-18+19.4+b3 [24.9 MB]
Get:19 http://ftp.us.debian.org/debian stretch/main amd64 gnat-mingw-w64-x86-64 amd64 6.3.0-18+19.4+b3 [25.8 MB]
Get:20 http://ftp.us.debian.org/debian stretch/main amd64 gnat-mingw-w64 all 6.3.0-14+19.4 [168 kB]
Get:21 http://ftp.us.debian.org/debian stretch/main amd64 mingw-w64 all 5.0.1-1 [12.3 kB]
Get:22 http://ftp.us.debian.org/debian stretch/main amd64 mingw-w64-tools amd64 5.0.1-1 [224 kB]
Fetched 209 MB in 5min 42s (610 kB/s)                                          
Selecting previously unselected package pkg-config.
(Reading database ... 301703 files and directories currently installed.)
Preparing to unpack .../00-pkg-config_0.29-4+b1_amd64.deb ...
Unpacking pkg-config (0.29-4+b1) ...
Selecting previously unselected package binutils-mingw-w64-i686.
Preparing to unpack .../01-binutils-mingw-w64-i686_2.28-5+7.4+b4_amd64.deb ...
Unpacking binutils-mingw-w64-i686 (2.28-5+7.4+b4) ...
Selecting previously unselected package binutils-mingw-w64-x86-64.
Preparing to unpack .../02-binutils-mingw-w64-x86-64_2.28-5+7.4+b4_amd64.deb ...
Unpacking binutils-mingw-w64-x86-64 (2.28-5+7.4+b4) ...
Selecting previously unselected package mingw-w64-common.
Preparing to unpack .../03-mingw-w64-common_5.0.1-1_all.deb ...
Unpacking mingw-w64-common (5.0.1-1) ...
Selecting previously unselected package mingw-w64-i686-dev.
Preparing to unpack .../04-mingw-w64-i686-dev_5.0.1-1_all.deb ...
Unpacking mingw-w64-i686-dev (5.0.1-1) ...
Selecting previously unselected package gcc-mingw-w64-base.
Preparing to unpack .../05-gcc-mingw-w64-base_6.3.0-18+19.3+b3_amd64.deb ...
Unpacking gcc-mingw-w64-base (6.3.0-18+19.3+b3) ...
Selecting previously unselected package gcc-mingw-w64-i686.
Preparing to unpack .../06-gcc-mingw-w64-i686_6.3.0-18+19.3+b3_amd64.deb ...
Unpacking gcc-mingw-w64-i686 (6.3.0-18+19.3+b3) ...
Selecting previously unselected package g++-mingw-w64-i686.
Preparing to unpack .../07-g++-mingw-w64-i686_6.3.0-18+19.3+b3_amd64.deb ...
Unpacking g++-mingw-w64-i686 (6.3.0-18+19.3+b3) ...
Selecting previously unselected package mingw-w64-x86-64-dev.
Preparing to unpack .../08-mingw-w64-x86-64-dev_5.0.1-1_all.deb ...
Unpacking mingw-w64-x86-64-dev (5.0.1-1) ...
Selecting previously unselected package gcc-mingw-w64-x86-64.
Preparing to unpack .../09-gcc-mingw-w64-x86-64_6.3.0-18+19.3+b3_amd64.deb ...
Unpacking gcc-mingw-w64-x86-64 (6.3.0-18+19.3+b3) ...
Selecting previously unselected package g++-mingw-w64-x86-64.
Preparing to unpack .../10-g++-mingw-w64-x86-64_6.3.0-18+19.3+b3_amd64.deb ...
Unpacking g++-mingw-w64-x86-64 (6.3.0-18+19.3+b3) ...
Selecting previously unselected package g++-mingw-w64.
Preparing to unpack .../11-g++-mingw-w64_6.3.0-14+19.3_all.deb ...
Unpacking g++-mingw-w64 (6.3.0-14+19.3) ...
Selecting previously unselected package gcc-mingw-w64.
Preparing to unpack .../12-gcc-mingw-w64_6.3.0-14+19.3_all.deb ...
Unpacking gcc-mingw-w64 (6.3.0-14+19.3) ...
Selecting previously unselected package gfortran-mingw-w64-i686.
Preparing to unpack .../13-gfortran-mingw-w64-i686_6.3.0-18+19.3+b3_amd64.deb ...
Unpacking gfortran-mingw-w64-i686 (6.3.0-18+19.3+b3) ...
Selecting previously unselected package gfortran-mingw-w64-x86-64.
Preparing to unpack .../14-gfortran-mingw-w64-x86-64_6.3.0-18+19.3+b3_amd64.deb ...
Unpacking gfortran-mingw-w64-x86-64 (6.3.0-18+19.3+b3) ...
Selecting previously unselected package gfortran-mingw-w64.
Preparing to unpack .../15-gfortran-mingw-w64_6.3.0-14+19.3_all.deb ...
Unpacking gfortran-mingw-w64 (6.3.0-14+19.3) ...
Selecting previously unselected package gnat-mingw-w64-base.
Preparing to unpack .../16-gnat-mingw-w64-base_6.3.0-18+19.4+b3_amd64.deb ...
Unpacking gnat-mingw-w64-base (6.3.0-18+19.4+b3) ...
Selecting previously unselected package gnat-mingw-w64-i686.
Preparing to unpack .../17-gnat-mingw-w64-i686_6.3.0-18+19.4+b3_amd64.deb ...
Unpacking gnat-mingw-w64-i686 (6.3.0-18+19.4+b3) ...
Selecting previously unselected package gnat-mingw-w64-x86-64.
Preparing to unpack .../18-gnat-mingw-w64-x86-64_6.3.0-18+19.4+b3_amd64.deb ...
Unpacking gnat-mingw-w64-x86-64 (6.3.0-18+19.4+b3) ...
Selecting previously unselected package gnat-mingw-w64.
Preparing to unpack .../19-gnat-mingw-w64_6.3.0-14+19.4_all.deb ...
Unpacking gnat-mingw-w64 (6.3.0-14+19.4) ...
Selecting previously unselected package mingw-w64.
Preparing to unpack .../20-mingw-w64_5.0.1-1_all.deb ...
Unpacking mingw-w64 (5.0.1-1) ...
Selecting previously unselected package mingw-w64-tools.
Preparing to unpack .../21-mingw-w64-tools_5.0.1-1_amd64.deb ...
Unpacking mingw-w64-tools (5.0.1-1) ...
Setting up binutils-mingw-w64-x86-64 (2.28-5+7.4+b4) ...
Setting up binutils-mingw-w64-i686 (2.28-5+7.4+b4) ...
Setting up gcc-mingw-w64-base (6.3.0-18+19.3+b3) ...
Setting up pkg-config (0.29-4+b1) ...
Processing triggers for man-db (2.7.6.1-2) ...
Setting up mingw-w64-tools (5.0.1-1) ...
Setting up gnat-mingw-w64-base (6.3.0-18+19.4+b3) ...
Setting up mingw-w64-common (5.0.1-1) ...
Setting up mingw-w64-i686-dev (5.0.1-1) ...
Setting up gcc-mingw-w64-i686 (6.3.0-18+19.3+b3) ...
update-alternatives: using /usr/bin/i686-w64-mingw32-gcc-posix to provide /usr/bin/i686-w64-mingw32-gcc (i686-w64-mingw32-gcc) in auto mode
update-alternatives: warning: skip creation of /usr/bin/i686-w64-mingw32-gcc-6 because associated file /usr/bin/i686-w64-mingw32-gcc-6-posix (of link group i686-w64-mingw32-gcc) doesn't exist
update-alternatives: using /usr/bin/i686-w64-mingw32-gcc-win32 to provide /usr/bin/i686-w64-mingw32-gcc (i686-w64-mingw32-gcc) in auto mode
update-alternatives: warning: skip creation of /usr/bin/i686-w64-mingw32-gcc-6 because associated file /usr/bin/i686-w64-mingw32-gcc-6-win32 (of link group i686-w64-mingw32-gcc) doesn't exist
Setting up mingw-w64-x86-64-dev (5.0.1-1) ...
Setting up gnat-mingw-w64-i686 (6.3.0-18+19.4+b3) ...
update-alternatives: using /usr/bin/i686-w64-mingw32-gnat-posix to provide /usr/bin/i686-w64-mingw32-gnat (i686-w64-mingw32-gnat) in auto mode
update-alternatives: using /usr/bin/i686-w64-mingw32-gnat-win32 to provide /usr/bin/i686-w64-mingw32-gnat (i686-w64-mingw32-gnat) in auto mode
Setting up g++-mingw-w64-i686 (6.3.0-18+19.3+b3) ...
update-alternatives: using /usr/bin/i686-w64-mingw32-g++-posix to provide /usr/bin/i686-w64-mingw32-g++ (i686-w64-mingw32-g++) in auto mode
update-alternatives: using /usr/bin/i686-w64-mingw32-g++-win32 to provide /usr/bin/i686-w64-mingw32-g++ (i686-w64-mingw32-g++) in auto mode
Setting up gfortran-mingw-w64-i686 (6.3.0-18+19.3+b3) ...
update-alternatives: using /usr/bin/i686-w64-mingw32-gfortran-posix to provide /usr/bin/i686-w64-mingw32-gfortran (i686-w64-mingw32-gfortran) in auto mode
update-alternatives: using /usr/bin/i686-w64-mingw32-gfortran-win32 to provide /usr/bin/i686-w64-mingw32-gfortran (i686-w64-mingw32-gfortran) in auto mode
Setting up gcc-mingw-w64-x86-64 (6.3.0-18+19.3+b3) ...
update-alternatives: using /usr/bin/x86_64-w64-mingw32-gcc-posix to provide /usr/bin/x86_64-w64-mingw32-gcc (x86_64-w64-mingw32-gcc) in auto mode
update-alternatives: warning: skip creation of /usr/bin/x86_64-w64-mingw32-gcc-6 because associated file /usr/bin/x86_64-w64-mingw32-gcc-6-posix (of link group x86_64-w64-mingw32-gcc) doesn't exist
update-alternatives: using /usr/bin/x86_64-w64-mingw32-gcc-win32 to provide /usr/bin/x86_64-w64-mingw32-gcc (x86_64-w64-mingw32-gcc) in auto mode
update-alternatives: warning: skip creation of /usr/bin/x86_64-w64-mingw32-gcc-6 because associated file /usr/bin/x86_64-w64-mingw32-gcc-6-win32 (of link group x86_64-w64-mingw32-gcc) doesn't exist
Setting up gfortran-mingw-w64-x86-64 (6.3.0-18+19.3+b3) ...
update-alternatives: using /usr/bin/x86_64-w64-mingw32-gfortran-posix to provide /usr/bin/x86_64-w64-mingw32-gfortran (x86_64-w64-mingw32-gfortran) in auto mode
update-alternatives: using /usr/bin/x86_64-w64-mingw32-gfortran-win32 to provide /usr/bin/x86_64-w64-mingw32-gfortran (x86_64-w64-mingw32-gfortran) in auto mode
Setting up gfortran-mingw-w64 (6.3.0-14+19.3) ...
Setting up gcc-mingw-w64 (6.3.0-14+19.3) ...
Setting up g++-mingw-w64-x86-64 (6.3.0-18+19.3+b3) ...
update-alternatives: using /usr/bin/x86_64-w64-mingw32-g++-posix to provide /usr/bin/x86_64-w64-mingw32-g++ (x86_64-w64-mingw32-g++) in auto mode
update-alternatives: using /usr/bin/x86_64-w64-mingw32-g++-win32 to provide /usr/bin/x86_64-w64-mingw32-g++ (x86_64-w64-mingw32-g++) in auto mode
Setting up gnat-mingw-w64-x86-64 (6.3.0-18+19.4+b3) ...
update-alternatives: using /usr/bin/x86_64-w64-mingw32-gnat-posix to provide /usr/bin/x86_64-w64-mingw32-gnat (x86_64-w64-mingw32-gnat) in auto mode
update-alternatives: using /usr/bin/x86_64-w64-mingw32-gnat-win32 to provide /usr/bin/x86_64-w64-mingw32-gnat (x86_64-w64-mingw32-gnat) in auto mode
Setting up g++-mingw-w64 (6.3.0-14+19.3) ...
Setting up gnat-mingw-w64 (6.3.0-14+19.4) ...
Setting up mingw-w64 (5.0.1-1) ...
root@debian:/home/debian# 
root@debian:/home/debian# sudo apt install openjdk-8-jdk
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  libice-dev libpthread-stubs0-dev libsm-dev libx11-dev libx11-doc libxau-dev libxcb1-dev
  libxdmcp-dev libxt-dev openjdk-8-jdk-headless x11proto-core-dev x11proto-input-dev
  x11proto-kb-dev xorg-sgml-doctools xtrans-dev
Suggested packages:
  libice-doc libsm-doc libxcb-doc libxt-doc openjdk-8-demo openjdk-8-source visualvm
The following NEW packages will be installed:
  libice-dev libpthread-stubs0-dev libsm-dev libx11-dev libx11-doc libxau-dev libxcb1-dev
  libxdmcp-dev libxt-dev openjdk-8-jdk openjdk-8-jdk-headless x11proto-core-dev
  x11proto-input-dev x11proto-kb-dev xorg-sgml-doctools xtrans-dev
0 upgraded, 16 newly installed, 0 to remove and 0 not upgraded.
Need to get 15.7 MB of archives.
After this operation, 62.0 MB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://ftp.us.debian.org/debian stretch/main amd64 xorg-sgml-doctools all 1:1.11-1 [21.9 kB]
Get:2 http://ftp.us.debian.org/debian stretch/main amd64 x11proto-core-dev all 7.0.31-1 [728 kB]
Get:3 http://security.debian.org/debian-security stretch/updates/main amd64 openjdk-8-jdk-headless amd64 8u242-b08-1~deb9u1 [8,245 kB]
Get:4 http://ftp.us.debian.org/debian stretch/main amd64 libice-dev amd64 2:1.0.9-2 [66.8 kB]
Get:5 http://ftp.us.debian.org/debian stretch/main amd64 libpthread-stubs0-dev amd64 0.3-4 [3,866 B]
Get:6 http://ftp.us.debian.org/debian stretch/main amd64 libsm-dev amd64 2:1.2.2-1+b3 [35.8 kB]
Get:7 http://ftp.us.debian.org/debian stretch/main amd64 libxau-dev amd64 1:1.0.8-1 [23.6 kB]
Get:8 http://ftp.us.debian.org/debian stretch/main amd64 libxdmcp-dev amd64 1:1.1.2-3 [42.2 kB]
Get:9 http://ftp.us.debian.org/debian stretch/main amd64 x11proto-input-dev all 2.3.2-1 [158 kB]
Get:10 http://ftp.us.debian.org/debian stretch/main amd64 x11proto-kb-dev all 1.0.7-1 [233 kB]
Get:11 http://ftp.us.debian.org/debian stretch/main amd64 xtrans-dev all 1.3.5-1 [100 kB]
Get:12 http://ftp.us.debian.org/debian stretch/main amd64 libxcb1-dev amd64 1.12-1 [169 kB]
Get:13 http://ftp.us.debian.org/debian stretch/main amd64 libx11-dev amd64 2:1.6.4-3+deb9u1 [815 kB]
Get:14 http://security.debian.org/debian-security stretch/updates/main amd64 openjdk-8-jdk amd64 8u242-b08-1~deb9u1 [2,396 kB]
Get:15 http://ftp.us.debian.org/debian stretch/main amd64 libx11-doc all 2:1.6.4-3+deb9u1 [2,201 kB]
Get:16 http://ftp.us.debian.org/debian stretch/main amd64 libxt-dev amd64 1:1.1.5-1 [423 kB]
Fetched 15.7 MB in 1min 10s (223 kB/s)                                                     
Selecting previously unselected package xorg-sgml-doctools.
(Reading database ... 321249 files and directories currently installed.)
Preparing to unpack .../00-xorg-sgml-doctools_1%3a1.11-1_all.deb ...
Unpacking xorg-sgml-doctools (1:1.11-1) ...
Selecting previously unselected package x11proto-core-dev.
Preparing to unpack .../01-x11proto-core-dev_7.0.31-1_all.deb ...
Unpacking x11proto-core-dev (7.0.31-1) ...
Selecting previously unselected package libice-dev:amd64.
Preparing to unpack .../02-libice-dev_2%3a1.0.9-2_amd64.deb ...
Unpacking libice-dev:amd64 (2:1.0.9-2) ...
Selecting previously unselected package libpthread-stubs0-dev:amd64.
Preparing to unpack .../03-libpthread-stubs0-dev_0.3-4_amd64.deb ...
Unpacking libpthread-stubs0-dev:amd64 (0.3-4) ...
Selecting previously unselected package libsm-dev:amd64.
Preparing to unpack .../04-libsm-dev_2%3a1.2.2-1+b3_amd64.deb ...
Unpacking libsm-dev:amd64 (2:1.2.2-1+b3) ...
Selecting previously unselected package libxau-dev:amd64.
Preparing to unpack .../05-libxau-dev_1%3a1.0.8-1_amd64.deb ...
Unpacking libxau-dev:amd64 (1:1.0.8-1) ...
Selecting previously unselected package libxdmcp-dev:amd64.
Preparing to unpack .../06-libxdmcp-dev_1%3a1.1.2-3_amd64.deb ...
Unpacking libxdmcp-dev:amd64 (1:1.1.2-3) ...
Selecting previously unselected package x11proto-input-dev.
Preparing to unpack .../07-x11proto-input-dev_2.3.2-1_all.deb ...
Unpacking x11proto-input-dev (2.3.2-1) ...
Selecting previously unselected package x11proto-kb-dev.
Preparing to unpack .../08-x11proto-kb-dev_1.0.7-1_all.deb ...
Unpacking x11proto-kb-dev (1.0.7-1) ...
Selecting previously unselected package xtrans-dev.
Preparing to unpack .../09-xtrans-dev_1.3.5-1_all.deb ...
Unpacking xtrans-dev (1.3.5-1) ...
Selecting previously unselected package libxcb1-dev:amd64.
Preparing to unpack .../10-libxcb1-dev_1.12-1_amd64.deb ...
Unpacking libxcb1-dev:amd64 (1.12-1) ...
Selecting previously unselected package libx11-dev:amd64.
Preparing to unpack .../11-libx11-dev_2%3a1.6.4-3+deb9u1_amd64.deb ...
Unpacking libx11-dev:amd64 (2:1.6.4-3+deb9u1) ...
Selecting previously unselected package libx11-doc.
Preparing to unpack .../12-libx11-doc_2%3a1.6.4-3+deb9u1_all.deb ...
Unpacking libx11-doc (2:1.6.4-3+deb9u1) ...
Selecting previously unselected package libxt-dev:amd64.
Preparing to unpack .../13-libxt-dev_1%3a1.1.5-1_amd64.deb ...
Unpacking libxt-dev:amd64 (1:1.1.5-1) ...
Selecting previously unselected package openjdk-8-jdk-headless:amd64.
Preparing to unpack .../14-openjdk-8-jdk-headless_8u242-b08-1~deb9u1_amd64.deb ...
Unpacking openjdk-8-jdk-headless:amd64 (8u242-b08-1~deb9u1) ...
Selecting previously unselected package openjdk-8-jdk:amd64.
Preparing to unpack .../15-openjdk-8-jdk_8u242-b08-1~deb9u1_amd64.deb ...
Unpacking openjdk-8-jdk:amd64 (8u242-b08-1~deb9u1) ...
Setting up libpthread-stubs0-dev:amd64 (0.3-4) ...
Setting up openjdk-8-jdk-headless:amd64 (8u242-b08-1~deb9u1) ...
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/idlj to provide /usr/bin/idlj (idlj) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/jdeps to provide /usr/bin/jdeps (jdeps) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/wsimport to provide /usr/bin/wsimport (wsimport) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/rmic to provide /usr/bin/rmic (rmic) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/jinfo to provide /usr/bin/jinfo (jinfo) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/jsadebugd to provide /usr/bin/jsadebugd (jsadebugd) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/native2ascii to provide /usr/bin/native2ascii (native2ascii) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/jstat to provide /usr/bin/jstat (jstat) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/javac to provide /usr/bin/javac (javac) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/javah to provide /usr/bin/javah (javah) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/clhsdb to provide /usr/bin/clhsdb (clhsdb) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/jstack to provide /usr/bin/jstack (jstack) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/jrunscript to provide /usr/bin/jrunscript (jrunscript) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/javadoc to provide /usr/bin/javadoc (javadoc) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/javap to provide /usr/bin/javap (javap) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/jar to provide /usr/bin/jar (jar) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/extcheck to provide /usr/bin/extcheck (extcheck) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/hsdb to provide /usr/bin/hsdb (hsdb) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/schemagen to provide /usr/bin/schemagen (schemagen) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/jps to provide /usr/bin/jps (jps) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/xjc to provide /usr/bin/xjc (xjc) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/jarsigner to provide /usr/bin/jarsigner (jarsigner) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/jmap to provide /usr/bin/jmap (jmap) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/jstatd to provide /usr/bin/jstatd (jstatd) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/jhat to provide /usr/bin/jhat (jhat) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/jdb to provide /usr/bin/jdb (jdb) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/serialver to provide /usr/bin/serialver (serialver) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/wsgen to provide /usr/bin/wsgen (wsgen) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/jcmd to provide /usr/bin/jcmd (jcmd) in auto mode
Setting up openjdk-8-jdk:amd64 (8u242-b08-1~deb9u1) ...
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/appletviewer to provide /usr/bin/appletviewer (appletviewer) in auto mode
update-alternatives: using /usr/lib/jvm/java-8-openjdk-amd64/bin/jconsole to provide /usr/bin/jconsole (jconsole) in auto mode
Setting up xorg-sgml-doctools (1:1.11-1) ...
Setting up x11proto-kb-dev (1.0.7-1) ...
Processing triggers for sgml-base (1.29) ...
Setting up xtrans-dev (1.3.5-1) ...
Setting up libx11-doc (2:1.6.4-3+deb9u1) ...
Processing triggers for man-db (2.7.6.1-2) ...
Setting up x11proto-core-dev (7.0.31-1) ...
Setting up libxau-dev:amd64 (1:1.0.8-1) ...
Setting up libxdmcp-dev:amd64 (1:1.1.2-3) ...
Setting up libice-dev:amd64 (2:1.0.9-2) ...
Setting up libxcb1-dev:amd64 (1.12-1) ...
Setting up x11proto-input-dev (2.3.2-1) ...
Setting up libsm-dev:amd64 (2:1.2.2-1+b3) ...
Setting up libx11-dev:amd64 (2:1.6.4-3+deb9u1) ...
Setting up libxt-dev:amd64 (1:1.1.5-1) ...
root@debian:/home/debian# cd Downloads
root@debian:/home/debian/Downloads# cd scr
root@debian:/home/debian/Downloads/scr# make -f Makefile.CrossWindows
bash: make: command not found

Reply:
root@debian:/home/debian/Downloads/scr/scr# make -f Makefile.CrossWindows
fatal: Not a git repository (or any of the parent directories): .git
./gradlew clean

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':server'.
> SDK location not found. Define location with an ANDROID_SDK_ROOT environment variable or by setting the sdk.dir path in your project's local properties file at '/home/debian/Downloads/scr/scr/local.properties'.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
Makefile.CrossWindows:40: recipe for target 'clean' failed
make: *** [clean] Error 1

Reply:
how to fix sdl2 error?

root@debian:/home/debian/Downloads/scr/scr# make -f Makefile.CrossWindows
fatal: Not a git repository (or any of the parent directories): .git
./gradlew clean

BUILD SUCCESSFUL in 1s
1 actionable task: 1 up-to-date
rm -rf "build-server" "build-win32" "build-win64" \
	   "build-win32-noconsole" "build-win64-noconsole" "dist"
[ -d "build-server" ] || ( mkdir "build-server" && \
	meson "build-server" \
		--buildtype release -Dcompile_app=false )
The Meson build system
Version: 0.53.2
Source dir: /home/debian/Downloads/scr/scr
Build dir: /home/debian/Downloads/scr/scr/build-server
Build type: native build
Project name: scrcpy
Project version: 1.12.1
C compiler for the host machine: cc (gcc 6.3.0 "cc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516")
C linker for the host machine: cc ld.bfd 2.28
Host machine cpu family: x86_64
Host machine cpu: x86_64
Program ./scripts/build-wrapper.sh found: YES (/bin/bash /home/debian/Downloads/scr/scr/server/./scripts/build-wrapper.sh)
WARNING: Project targeting '>= 0.37' but tried to use feature introduced in '0.48.0': console arg in custom_target
DEPRECATION: build_always is deprecated. Combine build_by_default and build_always_stale instead.
Build targets in project: 2
WARNING: Project specifies a minimum meson_version '>= 0.37' but uses features which were added in newer versions:
 * 0.48.0: {'console arg in custom_target'}

Found ninja-1.7.2 at /usr/bin/ninja
ninja -C "build-server"
ninja: Entering directory `build-server'
[0/1] Generating scrcpy-server with a custom command.
(not invoking gradle, since we are root)
make -C prebuilt-deps prepare-win32
make[1]: Entering directory '/home/debian/Downloads/scr/scr/prebuilt-deps'
make[1]: execvp: ./prepare-dep: Permission denied
Makefile:33: recipe for target 'prepare-sdl2' failed
make[1]: *** [prepare-sdl2] Error 127
make[1]: Leaving directory '/home/debian/Downloads/scr/scr/prebuilt-deps'
Makefile.CrossWindows:51: recipe for target 'prepare-deps-win32' failed
make: [prepare-deps-win32] Error 2 (ignored)
[ -d "build-win32" ] || ( mkdir "build-win32" && \
	meson "build-win32" \
		--cross-file cross_win32.txt \
		--buildtype release --strip -Db_lto=true \
		-Dcrossbuild_windows=true \
		-Dcompile_server=false \
		-Dportable=true )
The Meson build system
Version: 0.53.2
Source dir: /home/debian/Downloads/scr/scr
Build dir: /home/debian/Downloads/scr/scr/build-win32
Build type: cross build
Project name: scrcpy
Project version: 1.12.1
C compiler for the build machine: cc (gcc 6.3.0 "cc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516")
C linker for the build machine: cc ld.bfd 2.28
C compiler for the host machine: /usr/bin/i686-w64-mingw32-gcc (gcc 6.3.0 "i686-w64-mingw32-gcc (GCC) 6.3.0 20170516")
C linker for the host machine: /usr/bin/i686-w64-mingw32-gcc ld.bfd 2.28
Build machine cpu family: x86_64
Build machine cpu: x86_64
Host machine cpu family: x86
Host machine cpu: i686
Target machine cpu family: x86
Target machine cpu: i686

app/meson.build:46:4: ERROR: Include dir ../prebuilt-deps/SDL2-2.0.10/i686-w64-mingw32/include does not exist.

A full log can be found at /home/debian/Downloads/scr/scr/build-win32/meson-logs/meson-log.txt
Makefile.CrossWindows:54: recipe for target 'build-win32' failed
make: *** [build-win32] Error 1

Reply:
This is not related to the pull request. I don't know how to build for windows either. I know you should **not be building as root** though. Please ask your question in an appropriate thread so people can help you.

PS: You should also checkout the `dev` branch before you run the build since `--lock-video-orientation` is what you are after.

Thanks